### DISCUSSION TITLE
Feature request: add parameter to textadept.file_types.set_lexer() to set specific lexer from init.lua

### DISCUSSION THREAD

**Discussion by paaguti** [orbitalquark/textadept#242] [Ideas]
I was wondering if there is any way to add a new function or modify `textadept.file_types.set_lexer()` to add the possibility of passing the lexer directly instead of opening the dialog to select the lexer.

It would be very handy to add something in my init.lua to change the current lexer to lua in a buffer with 
```
textadept.file_types.set_lexer("lua")
```
without having to go through the dialog

**Comment by snoopy** [comment]
Maybe I am not understanding correctly what you want to do but can't you just do:

```lua
buffer:set_lexer('lua')
events.emit(events.UPDATE_UI, 1) -- for updating statusbar
```
and bind this to a key or use it in a function?

**Comment by paaguti** [comment]
Hi,

<blush> I was so much concentrating on the binding stuff that I forgot the
update_ui stuff </blush>

Thx a ton, /PA

On Fri, 22 Jul 2022 at 10:08, snoopy ***@***.***> wrote:

> Maybe I am not understanding correctly what you want to do but can't you
> just do:
>
> buffer:set_lexer('lua')
> events.emit(events.UPDATE_UI, 1) -- for updating statusbar
>
> and bind this to a key or use it in a function?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/242#discussioncomment-3204205>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4GFCMFU3RD4DLE6773VVJJJVANCNFSM54KL2RMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Comment by orbitalquark** [comment]
I've committed a change that allows you to simply call `buffer:set_lexer()` without manually updating the UI. Thanks for catching this @snoopy.

https://github.com/orbitalquark/textadept/commit/8034c1e1c9aef39393062b56120b583a1e2a31ed

**Comment by paaguti** [comment]
Shouldn't that be done at the end for buffer:set_lexer() ?
Would be nice to avoid the update_ui stuff in my code ;-)

Mi .002 cents, /PA

On Sat, 23 Jul 2022 at 01:13, orbitalquark ***@***.***> wrote:

> I've committed a change that allows you to simply call
> `buffer:set_lexer()` without manually updating the UI. Thanks for catching
> this @snoopy.
>
>
> https://github.com/orbitalquark/textadept/commit/8034c1e1c9aef39393062b56120b583a1e2a31ed
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/242#discussioncomment-3210072>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4B3KW3UCUDNRW4QJJ3VVMTJXANCNFSM54KL2RMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Comment by orbitalquark** [comment]
Yes. Normally the `buffer:colorize()` call emits `events.UPDATE_UI` due to styling changes, but in the event there was no text in the buffer, `events.UPDATE_UI` wasn't happening. The change I committed now handles this case.
